### DISCUSSION TITLE
N64 saves

### DISCUSSION THREAD

**Discussion by capnsbum10** [sanni/cartreader#976] [Need help]
### OSCR firmware version

13.1

### OSCR hardware version

HW3

### System used

N64 Cartridge

### Describe the issue.

I’m trying to read the save data to copy to a duplicate game cartridge. When I go to read save data it says error no data received. Ive made sure the clock was defined and had clock 1 switch on

**Comment by sanni** [comment]
Can you open the N64 cartridge an post an image of the small 8-pin eeprom chip? Especially what is written on it. Then I can look if I have one with the same markings to check it out on my side.

You can also try [V10.2](https://github.com/sanni/cartreader/releases/tag/V10.2) which uses the old eeprom dump code. (CLK switch set to on.)
Or since you are on HW3 [V7.2](https://github.com/sanni/cartreader/wiki/Cart-Reader-V3#older-releasesfor-testing-newer-releases-will-work-with-v3-hardware-too) which uses an even older code that needs the CLK switch set to **off**.

**Comment by capnsbum10** [comment]
Waiting on screwdrivers to open the carts. I downgraded to version 10.2 and now it says reading eeprom… but hasn’t done anything else, appears to be frozen there.

**Comment by capnsbum10** [comment]
Tried to go down to firmware 7.3 but system warned of low memory availability and stability problems. Now the cart reader won’t turn on

**Comment by sanni** [comment]
In options.h you need to change:

````
//******************************************
// CHOOSE HARDWARE VERSION
//******************************************
#define HW4
//#define HW3
````
to :
````
//******************************************
// CHOOSE HARDWARE VERSION
//******************************************
//#define HW4
#define HW3
````

So that it runs on HW3. Then upload again.

**Reply by FatBeard** [reply]
Hi, 

I'm having similar issues with Super Mario 64 Pal. I can read the Rom but not the save. As stated by op I get "Error: no data received" when trying to do so. I can read my Zelda ocarina save. But it's sram as opposed to eeprom in the Mario cart. If you need me to provide pictures of the chips I can do so but would assume this is a popular cart and its internals are well known. I have a hw5 reader. It was bought from aliexpress with a 2.54mm pin pitch, however I made sure to align the cart properly to avoid crossing the pins. Any help would be appreciated.

**Reply by sanni** [reply]
Yes a picture would help, there is always the possibility of an unknown chip that simply wasn't tested yet.

What you also can try is replacing the Clock Generator with [one from Adafruit](https://www.adafruit.com/product/2045) as there have been [reports of bad Si5351 chips](https://www.eevblog.com/forum/projects/reputable-source-for-si5351a-clock-pll/) lately that behave erratic. (This is what I would do first, as I always had some Clock Generators that work better than others even before they were cloned)

You can also calibrate the [Clock Generator with the OSCR](https://github.com/sanni/cartreader/wiki/Calibrating-the-ClockGen) which might help if the Clock Gen works but is out of calibration.

If you are more technical minded you can buy a [logic analyzer](https://www.aliexpress.com/item/1005004526978203.html) and check what the eep data and eep clk pins are exactly doing. 

![eeprom_read](https://github.com/sanni/cartreader/assets/19169494/38348469-822e-452c-8020-62873a2d6c32)

**Reply by FatBeard** [reply]
Thanks for the reply. 

Here is a picture of the chips. 
![image](https://github.com/sanni/cartreader/assets/4692143/fcc49880-b4e4-4f79-a34d-fb187a9777c8)

It may actually be the clock though as when I attempted to calibrate all I get is "Read correction: 0" showing on the display as soon as I click on "Calibrate Clock" from sfc sub menu.  After which i attempted to connect pins 43 and 56 but nothing happend. (I'm running 13.5 firmware). The self test for the clock does pass though. 
![image](https://github.com/sanni/cartreader/assets/4692143/5f8b935c-414a-4767-b63d-009e65cb7997)

I actually have a logic analyser. I'll take a look look maybe this evening. But you're possibly correct in me needing to get a new clock breakout from adafruit.

**Reply by capnsbum10** [reply]
I changed it and the device functions, but when I select game cartridge under 64 it comes up with the following.

gamepak error
 Name: donkeykong64
Id: ndoe 

cartridge unknown

**Reply by capnsbum10** [reply]
![image](https://github.com/sanni/cartreader/assets/172704427/c4822879-0ef9-49cc-a5d7-44a1fb21beb5)

**Reply by capnsbum10** [reply]
![IMG_0615](https://github.com/sanni/cartreader/assets/172704427/9bf1ede2-1272-4f98-aac3-c3d16c755dbd)
![IMG_0616](https://github.com/sanni/cartreader/assets/172704427/22b6328f-cae9-464e-88f4-b23b3ae03094)


The chips for both Silicon Valley and donkey Kong

**Reply by FatBeard** [reply]
Was away the past week so didn't get a chance to do this. Poked my analyser out from a 5 year retirement. This is what i'm reading on the N64 eep clock pins. It's been a while since i've used this software and i wasn't an expert then, but seems a bit erratic? I'll try and capture some data tomorrow. 

<img width="781" alt="image" src="https://github.com/user-attachments/assets/57f5ea2c-b298-48cc-99ec-4f9c8135ffb8">

**Reply by FatBeard** [reply]
A new clock generator worked. Thank you
